### PR TITLE
sched/note: correct flatten format

### PR DESCRIPTION
**Pull request by anchao**
## Summary

sched/note: correct flatten format

## Impact
Fix endian issue reported by @pkarashchenko :
https://github.com/apache/incubator-nuttx-apps/pull/1114#discussion_r842466352

## Testing

sched note trace test

### CODE REVIEW

File: sched/sched/sched_note.c [hunk 1]
```diff
@@ -122,18 +123,17 @@ static inline void sched_note_flatten(FAR uint8_t *dst,
     {
 #ifdef CONFIG_HAVE_LONG_LONG
       case 8:
-        dst[7] = (uint8_t)((*(uint64_t *)src >> 56) & 0xff);
-        dst[6] = (uint8_t)((*(uint64_t *)src >> 48) & 0xff);
-        dst[5] = (uint8_t)((*(uint64_t *)src >> 40) & 0xff);
-        dst[4] = (uint8_t)((*(uint64_t *)src >> 32) & 0xff);
+        *(uint64_t *)dst = htole64(*(uint64_t *)src);
```
Comment (pkarashchenko) [inline sched/sched/sched_note.c:126]: Are we sure that `dst` points to properly aligned memory?

**Review by xiaoxiang781216 — APPROVED** [review]

File: sched/sched/sched_note.c [hunk 2]
```diff
@@ -116,24 +117,35 @@ static unsigned int g_note_disabled_irq_nest[CONFIG_SMP_NCPUS];
  ****************************************************************************/
 
 static inline void sched_note_flatten(FAR uint8_t *dst,
-                                      FAR void *src, size_t len)
+                                      FAR void *src_, size_t len)
 {
+  FAR uint8_t *src = src_;
+
   switch (len)
```
Comment (pkarashchenko) [inline sched/sched/sched_note.c:124]: Code is fine, but I was just thinking why just not to:
```
switch (len)
  {
    case 8:
    case 4:
    case 2:
    case 1:
#ifdef CONFIG_ENDIAN_BIG
      {
        FAR uint8_t *end = (FAR uint8_t *)src + len - 1;
        while (len-- > 0)
          {
            *dst++ = *end--;
          }
      }
#else
      memcpy(dst, src, len);
#endif
      break;
    default:
      DEBUGASSERT(FALSE);
      break;
  }
```
Seems to be more clear.

Comment (anchao) [reply sched/sched/sched_note.c:121]: Done

File: sched/sched/sched_note.c [hunk 3]
```diff
@@ -120,20 +120,21 @@ static inline void sched_note_flatten(FAR uint8_t *dst,
...
-        dst[4] = (uint8_t)((*(uint64_t *)src >> 32) & 0xff);
-#endif
       case 4:
-        dst[3] = (uint8_t)((*(uint32_t *)src >> 24) & 0xff);
-        dst[2] = (uint8_t)((*(uint32_t *)src >> 16) & 0xff);
       case 2:
-        dst[1] = (uint8_t)((*(uint16_t *)src >> 8) & 0xff);
       case 1:
-        dst[0] = (uint8_t)(*(uint8_t *)src & 0xff);
+#ifdef CONFIG_ENDIAN_BIG
```
Comment (xiaoxiang781216) [inline sched/sched/sched_note.c:127]: let's remove switch

Comment (pkarashchenko) [reply sched/sched/sched_note.c:127]: Switch partially ensures that we are handling standard types only.

Comment (xiaoxiang781216) [reply sched/sched/sched_note.c:127]: But it's just an internal function, we can trust caller.

Comment (pkarashchenko) [reply sched/sched/sched_note.c:127]: makes sense

Comment (pkarashchenko) [reply sched/sched/sched_note.c:127]: we can leave `DEBUGASSERT(len == 1 || len == 2 || len == 4 || len == 8);` optionally.

**Review by xiaoxiang781216 — APPROVED** [review]

**Review by pkarashchenko — APPROVED** [review]

**Pull request merged**